### PR TITLE
Add isPowerSaveMode method to ShadowPowerManager.

### PR DESCRIPTION
**Pull request by gruszczy**

### CODE REVIEW

**Comment by erd** [comment]
Please add some tests.

**Comment by gruszczy** [comment]
Hi Erich,

I added a test and I think I did it exactly like you did for isInteractive, but it doesn't seem to work. When I run mvn package it fails and it seems it still goes to the real PowerManager, rather than the shadow:

java.lang.NullPointerException: null
	at android.os.PowerManager.isPowerSaveMode(PowerManager.java:792)
	at org.robolectric.shadows.ShadowPowerManagerTest.isPowerSaveMode_shouldGetAndSet(ShadowPowerManagerTest.java:121)

Any idea what I am doing wrong? Am I missing some step when building?

**Comment by gruszczy** [comment]
Also, it seems there is something wrong again with the commits again. I made a change, amended the commit and when I wanted to push it again, it forced me to pull and merge first. Any suggestion what I am doing wrong?

**Comment by erd** [comment]
> I added a test and I think I did it exactly like you did for isInteractive, but it doesn't seem to work. When I run mvn package it fails and it seems it still goes to the real PowerManager, rather than the shadow

Robolectric's tests runs against the locally installed jars. You need to run `mvn install` from `robolectric-shadows/shadows-core` and then run `mvn test` from the root directory.

> Also, it seems there is something wrong again with the commits again. I made a change, amended the commit and when I wanted to push it again, it forced me to pull and merge first. Any suggestion what I am doing wrong?

If you are pushing to a feature branch, you can just 'git push -f' to overwrite the remote branch with your new changes. Use this with caution, as it will nuke the remote branch. GitHub will automatically update the PR, and Travis will build the new change set.

File: robolectric/src/test/java/org/robolectric/shadows/ShadowPowerManagerTest.java [hunk 1]
```diff
@@ -115,4 +115,11 @@ public void isInteractive_shouldGetAndSet() {
     shadowPowerManager.setIsInteractive(false);
     assertThat(powerManager.isInteractive()).isFalse();
   }
+
+  @Test @Config(emulateSdk = Build.VERSION_CODES.LOLLIPOP)
```
Comment (erd) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowPowerManagerTest.java]: Robolectric always runs the tests with the latest API version that's supported. You shouldn't need to specify `emulateSdk` unless you're targeting an older API level.

**Comment by gruszczy** [comment]
Thanks a lot for all advice. I installed everything and successfully run tests. I removed the line you asked about and repushed my changed -f for a clean commit. PTAL.

**Comment by erd** [comment]
Awesome, thanks!

**Comment by gruszczy** [comment]
Thanks a lot for patience and guidance!

**Pull request merged**